nav2-usability/scenario-code/dynamic-linking/dynamic_linking_vrouter.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Dynamic linking example
+/// Done using VRouter
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:vrouter/vrouter.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistApp extends StatefulWidget {
+  @override
+  _WishlistAppState createState() => _WishlistAppState();
+}
+
+class _WishlistAppState extends State<WishlistApp> {
+  final AppState _appState = AppState();
+  final vRouterKey;
+
+  _WishlistAppState() : vRouterKey = GlobalKey<VRouterState>() {
+    _appState.addListener(() {
+      setState(() {});
+    });
+  }
+
+  @override
+  void dispose() {
+    _appState.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      key: vRouterKey,
+      routes: [
+        VWidget(
+          path: '/',
+          widget: WishlistListScreen(wishlists: _appState.wishlists, onCreate: onCreate),
+          stackedRoutes: [
+            VWidget(
+              path: r'wishlist/:id(\d+)',
+              widget: Builder(
+                builder: (context) => WishlistScreen(
+                    wishlist: _appState.wishlists.firstWhere(
```

Comment:
what if the wishlist id is invalid?

Reply:
I can add a `VGuard` but since @johnpryan made [this](https://github.com/flutter/uxr/pull/28#discussion_r603684670) comment I feel like it is kind of the same issue. I can add it if you prefer though.

Reply:
I think we need to do that for this sample in order to demo the dynamic linking scenario. What do you think @johnpryan

Reply:
I see in was a requirement, done in 389e65682496ee90c7bdf8ed74a3862c28ed615d